### PR TITLE
Check that input and output names are different

### PR DESCRIPTION
**Pull request by ekushele**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/gnu/sort/main.nf [hunk 1]
```diff
@@ -23,7 +23,7 @@ process GNU_SORT {
     suffix          = task.ext.suffix   ?: "${input.extension}"
     output_file     = "${prefix}.${suffix}"
     def VERSION     = "9.1"             // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
-
+    if ("$input" == "$output_file") error "Input and output names are the same, use \"task.ext.prefix\" to disambiguate!"
```
Comment (nvnieuwk) [inline modules/nf-core/gnu/sort/main.nf:26]: Can you also add this to the stub?

**Review by nvnieuwk — CHANGES_REQUESTED** [review]

**Review by nvnieuwk — APPROVED** [review]
LGTM!!

**Comment by nvnieuwk** [comment]
Only linting fixes left to do

**Comment by ekushele** [comment]
@nvnieuwk fixed linting issues

**Review by nvnieuwk — APPROVED** [review]

**Pull request merged**